packages/@react-aria/interactions/src/usePress.ts

```diff
@@ -107,6 +107,16 @@ export function usePress(props: PressHookProps): PressResult {
     isOverTarget: false
   });
 
+  let globalListeners = useRef(new Map());
+  let addGlobalListener = useCallback((eventTarget, type, listener, options) => {
```

Comment:
Do we have to worry about race conditions and the order of these? I noticed that the add and the remove do things in reverse order.

Reply:
I don't think the order should matter there. JS is single threaded so nothing will run between the function calls.